environment/container/dockerengine.go

```diff
@@ -146,34 +172,31 @@ func (e *dockerEngine) InspectImage(image string) (types.ImageInspect, error) {
 }
 
 // CreateContainer creates a container
-func (e *dockerEngine) CreateContainer(image string) (containerid string, err error) {
-	if err := e.pullImage(image); err != nil {
-		return "", fmt.Errorf("failed to pull the image '%s'. Error: %q", image, err)
+func (e *dockerEngine) CreateContainer(container environmenttypes.Container) (containerid string, err error) {
+	if err := e.pullImage(container.Image); err != nil {
+		return "", fmt.Errorf("failed to pull the image '%s'. Error: %w", container.Image, err)
+	}
+	contconfig := &containertypes.Config{
+		Image: container.Image,
 	}
-	contconfig := &container.Config{
-		Image: image,
-		Cmd:   []string{"sh", "-c", "tail -f /dev/null"},
+	if len(container.Command) > 0 {
+		contconfig.Cmd = container.Command
```

Comment:
```
We need to somehow indicate to the user that container.Command should be a command which will keep the container active and not something which will start and stop.
```
? No that's not how it works.
It's just a command to override the entrypoint same as Docker compose https://docs.docker.com/compose/compose-file/#command or Kubernetes https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.19/#container-v1-core

There's no guarantee that it will keep the container running.

Reply:
The way we utilise the command, if it is not persistent the container will not be there when the detect and transform commands execute. That's because we expect a container to be running and we do a exec for the detect and transform commands.